### DISCUSSION TITLE
How to deploy turborepo kitchen-sink example express app in vercel?

### DISCUSSION THREAD

**Discussion by msaglietto** [vercel/turborepo#8014] [Help] [answered]
### Summary

Hello,

Sorry if I repeat this question https://github.com/vercel/turbo/discussions/1479 but it seems now vercel has a way to deploy express https://vercel.com/guides/using-express-with-vercel

But for the kitchen-sink approach, I can't make it work and I don't know what I'm making different

So the kitchen sink has the app/api folder so I go to that folder and create a vercel.json with:
```
{ "version": 2, "rewrites": [{ "source": "/(.*)", "destination": "/src" }] }
```
then modify the  src/index.ts to return the express like in the express example here https://github.com/vercel/examples/blob/main/solutions/express/api/index.ts

Also, modify the package.json with the
```  
  "main": "index.ts",
  "scripts": {
    "start": "node src/index.ts",
```
So when I deploy that it returns the usual error 
```
Error: No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your Project Settings.
```
so I do an override of output to `.`  strange because the express example above doesn't do any override and is converting  . to the public folder 

When you do the override to `.` instead of executing the function it wants to download the src folder

So for the express example it detects a function during the deploy:
![image](https://github.com/vercel/turbo/assets/441786/54716806-a0bb-4021-b7ed-d3a9b36ce06f)
But not for the kitchen sink express =( 

Any pointers on what I'm doing wrong?

### Additional information

```js
Kitchen sink: https://turbo-test-cxrctebb9-mauricios-projects-f7bd08b2.vercel.app
Express sample: https://express-test-6jwf99euk-mauricios-projects-f7bd08b2.vercel.app
```

**Comment by herr-ahman** [comment]
We're running into the exact same issues... Have you solved it somehow @msaglietto?

**Comment by msaglietto** [comment, marked as answer]
We managed to make it work by renaming the api src folder to api and
exporting the express app on the index..
Not sure if we have more changes but basically vercel needs the folder to
be called api to detect functions (a bit nonsense if you ask me)
